scribe/lib/scribe/ai/presentation/widgets/ai_scribe_menu.dart

```diff
@@ -0,0 +1,250 @@
+
+class AIScribeMenu extends StatefulWidget {
+  final Function(AIScribeMenuAction) onActionSelected;
+  final bool useSubmenuItemStyle;
+  final List<AIScribeMenuCategory>? availableCategories;
+
+  const AIScribeMenu({
+    super.key,
+    required this.onActionSelected,
+    this.useSubmenuItemStyle = true,
+    this.availableCategories,
+  });
+
+  @override
+  State<AIScribeMenu> createState() => _AIScribeMenuContentState();
+}
+
+class _AIScribeMenuContentState extends State<AIScribeMenu> {
+  AIScribeMenuCategory? _hoveredCategory;
+  final Map<AIScribeMenuCategory, GlobalKey> _categoryKeys = {};
+  final GlobalKey _menuKey = GlobalKey();
+  OverlayEntry? _submenuOverlay;
+
+  @override
+  void initState() {
+    super.initState();
+    final categories = widget.availableCategories ?? AIScribeMenuCategory.values;
+    for (final category in categories) {
+      _categoryKeys[category] = GlobalKey();
+    }
+  }
+
+  @override
+  void dispose() {
+    _removeSubmenu();
+    super.dispose();
+  }
+
+  void _removeSubmenu() {
+    _submenuOverlay?.remove();
+    _submenuOverlay = null;
+  }
+
+  void _showSubmenu(AIScribeMenuCategory category) {
+    if (!category.hasSubmenu) return;
+
+    _removeSubmenu();
+
+    // Get the entire menu's bounds
+    final menuRenderBox = _menuKey.currentContext?.findRenderObject() as RenderBox?;
+    if(menuRenderBox == null) return;
+
+    final menuPosition = menuRenderBox.localToGlobal(Offset.zero);
+    final menuSize = menuRenderBox.size;
+
+    _submenuOverlay = OverlayEntry(
+      builder: (context) => _SubmenuPanel(
+        category: category,
+        position: menuPosition,
+        parentSize: menuSize,
+        onActionSelected: (action) {
+          _removeSubmenu();
+          widget.onActionSelected(action);
+        },
+        onDismiss: () {
+          setState(() {
+            _hoveredCategory = null;
+          });
+          _removeSubmenu();
+        },
+      ),
+    );
+
+    Overlay.of(context).insert(_submenuOverlay!);
+  }
+
+  void _handleCategoryClick(AIScribeMenuCategory category) {
+    setState(() {
+      // Toggle submenu - close if already open, open if closed
+      if (_hoveredCategory == category) {
+        _hoveredCategory = null;
+        _removeSubmenu();
+      } else {
+        _hoveredCategory = category;
+        _showSubmenu(category);
+      }
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final categories = widget.availableCategories ?? AIScribeMenuCategory.values;
+    return Column(
+      key: _menuKey,
+      mainAxisSize: MainAxisSize.min,
+      crossAxisAlignment: CrossAxisAlignment.stretch,
+      children: categories.map((category) {
+        return _buildCategoryItem(category);
+      }).toList(),
+    );
+  }
+
+  Widget _buildCategoryItem(AIScribeMenuCategory category) {
+    if (category.hasSubmenu) {
+      return Container(
+        key: _categoryKeys[category],
+        child: _buildMenuItem(
```

Comment:
Why not just pass the key directly to _buildMenuItem?

Reply:
Actually it has no purpose anymore and I forgot to remove it during refactoring and commit merging. I removed it.